File: 20220720/모법답안/20220720_01.js/StringToAscii_test.js
import assert from 'assert'
import solution from './StringToAscii.js'

describe('Custom Test', function () {
  it('test 1', function (done) {
    // given
    const s = 'work'

    // when
    const result = solution(s)

    // then
    assert.deepStrictEqual([119, 111, 114, 107], result)
    done();
  });

  it('test 2', function (done) {
    // given
    const s = 'Google'

    // when
    const result = solution(s)

    // then
    assert.deepStrictEqual([71, 111, 111, 103, 108, 101], result)
    done();
  });

  it('test 3', function (done) {
    // given
    const s = 'AlGoRitHm'

    // when
    const result = solution(s)

    // then
    assert.deepStrictEqual([65, 108, 71, 111, 82, 105, 116, 72, 109], result)
    done();
  });

  it('test 4', function (done) {
    // given
    const s = ''

    // when
    const result = solution(s)

    // then
    assert.deepStrictEqual([], result)
    done();
  });

  it('test 5', function (done) {
    // given
    const s = 'A'

    // when
    const result = solution(s)

    // then
    assert.deepStrictEqual([65], result)
    done();
  });

  it('test 6', function (done) {
    // given
    const s = 'z'

    // when
    const result = solution(s)

    // then
    assert.deepStrictEqual([122], result)
    done();
  });
})

File: 20220720/모법답안/20220720_01.js/StringToAscii_test.py
import unittest

from src.com.luckydave.algorithm.fastcampus.supereasy.StringToAscii.StringToAscii import solution


class CustomTestCase(unittest.TestCase):

    def test_1(self):
        # given
        s = 'work'

        # when
        result = solution(s)

        # then
        self.assertEqual([119, 111, 114, 107], result)

    def test_2(self):
        # given
        s = 'Google'

        # when
        result = solution(s)

        # then
        self.assertEqual([71, 111, 111, 103, 108, 101], result)

    def test_3(self):
        # given
        s = 'AlGoRitHm'

        # when
        result = solution(s)

        # then
        self.assertEqual([65, 108, 71, 111, 82, 105, 116, 72, 109], result)

    def test_4(self):
        # given
        s = ''

        # when
        result = solution(s)

        # then
        self.assertEqual([], result)

    def test_5(self):
        # given
        s = 'A'

        # when
        result = solution(s)

        # then
        self.assertEqual([65], result)

    def test_6(self):
        # given
        s = 'z'

        # when
        result = solution(s)

        # then
        self.assertEqual([122], result)


if __name__ == '__main__':
    unittest.main()
